### PR TITLE
Fix 2 issues when triggering a garbage collection via EventPipe

### PR DESCRIPTION
**Pull request by chrisnas**
Fix the following issues:
- https://github.com/dotnet/runtime/issues/99487: avoid double GC
- https://github.com/dotnet/runtime/issues/102572: handle client sequence number

I've tested with dotnet-gcstats (for the client sequence number + number of GCs) and dotnet-fullgc (to trigger a GC with a custom client sequence number)

@noahfalk / @brianrob: I'm not sure to understand why the enabled/disabled state was based on the number of session being > 0 because this number is always >= 1 when the functions are called.

### CODE REVIEW

**Comment by tommcdon** [comment]
@noahfalk @brianrob ptal

**Comment by brianrob** [comment]
Thanks @chrisnas!  One question - can you also test this with ETW?  I suspect that the ETW runtime is going to set `FilterData` differently.  I'll also have to defer to @noahfalk on the EventPipe specific file.

**Comment by chrisnas** [comment]
> Thanks @chrisnas! One question - can you also test this with ETW? I suspect that the ETW runtime is going to set `FilterData` differently. 

I did not try @brianrob...
Do you know the command line I should use with perfview?

File: src/native/eventpipe/ep.c [hunk 1]
```diff
@@ -673,6 +673,7 @@ disable_helper (EventPipeSessionID id)
 
 		while (ep_provider_callback_data_queue_try_dequeue (provider_callback_data_queue, &provider_callback_data)) {
 			ep_rt_prepare_provider_invoke_callback (&provider_callback_data);
+			provider_callback_data.enabled = false;
```
Comment (noahfalk) [inline src/native/eventpipe/ep.c:676]: Lets not modify enabled here. EventPipe is following a convention established by ETW. Modifying when we emit the PROVIDER_ENABLED control code makes ETW/EventPipe behave differently which makes it more difficult for anyone to write logic in a provider callback that works correctly for both.

**Review by noahfalk — CHANGES_REQUESTED** [review]
I'm happy to take a PR to fix this, but we need to modify the approach. Suggestions inline. Thanks!

**Comment by tommcdon** [comment]
Hi @chrisnas! I am moving this PR to "draft" mode - please let us know if you plan on addressing feedback in this PR.

**Comment by chrisnas** [comment]
> Hi @chrisnas! I am moving this PR to "draft" mode - please let us know if you plan on addressing feedback in this PR.

Sorry for the delay @tommcdon!
It's good to move it to draft because it seems that I underestimate the impact of the change for ETW.
When @noahfalk will have time, I will need to validate what should be done (especially defining the convention of exchanges between a tool and the runtime)

**Pull request closed without merging**